impl/maven-impl/src/main/java/org/apache/maven/impl/DefaultSettingsBuilder.java

```diff
@@ -237,23 +233,14 @@ private Settings readSettings(
     }
 
     private List<Server> serversByIds(List<Server> servers) {
-
-        if (servers.stream().allMatch(server -> server.getIds().isEmpty())) {
-            return servers;
-        }
-
         return servers.stream()
-                .flatMap(server -> {
-                    List<String> ids = server.getIds();
-                    if (ids.isEmpty()) {
-                        return Stream.of(server);
-                    }
-                    return Stream.concat(Stream.of(server), ids.stream().map(id -> Server.newBuilder(server, true)
-                            .id(id)
-                            .ids(Collections.emptyList())
-                            .build()));
-                })
-                .collect(Collectors.toList());
+                .flatMap(server -> Stream.concat(
+                        Stream.of(server), server.getIds().stream().map(id -> serverAlias(server, id))))
+                .toList();
+    }
+
+    private Server serverAlias(Server server, String id) {
+        return Server.newBuilder(server, true).id(id).ids(List.of()).build();
```

Comment:
```suggestion
        return newBuilder(server, true).id(id).build();
```
Server context / type is clear as directly located and its not generic `of` method, allowing static import.

Reply:
the same as below